aiogram/types/chat_member.py

```diff
@@ -31,53 +32,33 @@ class ChatMember(base.TelegramObject):
     can_send_other_messages: base.Boolean = fields.Field()
     can_add_web_page_previews: base.Boolean = fields.Field()
 
-    def is_admin(self):
-        warnings.warn('`is_admin` method deprecated due to updates in Bot API 4.2. '
-                      'This method renamed to `is_chat_admin` and will be available until aiogram 2.3',
-                      DeprecationWarning, stacklevel=2)
-        return self.is_chat_admin()
+    def is_chat_admin(self) -> bool:
+        return ChatMemberStatus.is_chat_admin(self.status)
 
-    def is_chat_admin(self):
-        return ChatMemberStatus.is_admin(self.status)
+    def is_chat_member(self) -> bool:
+        return ChatMemberStatus.is_chat_member(self.status, self.is_member)
 
-    def is_chat_member(self):
-        return ChatMemberStatus.is_member(self.status)
-
-    def __int__(self):
+    def __int__(self) -> int:
         return self.user.id
 
 
 class ChatMemberStatus(helper.Helper):
     """
     Chat member status
     """
-
     mode = helper.HelperMode.lowercase
 
     CREATOR = helper.Item()  # creator
     ADMINISTRATOR = helper.Item()  # administrator
     MEMBER = helper.Item()  # member
+    RESTRICTED = helper.Item()  # restricted
     LEFT = helper.Item()  # left
     KICKED = helper.Item()  # kicked
 
     @classmethod
-    def is_admin(cls, role):
-        warnings.warn('`is_admin` method deprecated due to updates in Bot API 4.2. '
-                      'This method renamed to `is_chat_admin` and will be available until aiogram 2.3',
-                      DeprecationWarning, stacklevel=2)
-        return cls.is_chat_admin(role)
-
-    @classmethod
-    def is_member(cls, role):
-        warnings.warn('`is_member` method deprecated due to updates in Bot API 4.2. '
-                      'This method renamed to `is_chat_member` and will be available until aiogram 2.3',
-                      DeprecationWarning, stacklevel=2)
-        return cls.is_chat_member(role)
-
-    @classmethod
-    def is_chat_admin(cls, role):
+    def is_chat_admin(cls, role: str) -> bool:
         return role in [cls.ADMINISTRATOR, cls.CREATOR]
 
     @classmethod
-    def is_chat_member(cls, role):
-        return role in [cls.MEMBER, cls.ADMINISTRATOR, cls.CREATOR]
+    def is_chat_member(cls, role: str, is_member: Optional[bool] = None) -> bool:
```

Comment:
Why is the new argument added?

Reply:
> Why is the new argument added?

Motivation: [aiogram_ru/67478](https://t.me/aiogram_ru/67478)

is_member in docs: 
> Restricted only. True, if the user is a member of the chat at the moment of the request

I don't know can a restricted user not be a chat member. If can't, a new argument is not needed.

Reply:
This case should be tested manually. I think new argument is not needed.

Reply:
This argument is not needed